src/deploy/osp_deployer/director.py

```diff
@@ -1520,41 +1523,60 @@ def run_sanity_test(self):
             pass
 
     def configure_tempest(self):
-        logger.debug("configuring tempest")
+        logger.info("Configuring tempest")
         setts = self.settings
+        external_sub_cmd = ("source ~/" + setts.overcloud_name + "rc;" +
+                            "openstack subnet list  | grep external_sub " +
+                            "| awk '{print $6;}'")
+        external_sub_guid = self.run_tty(external_sub_cmd)[0].rstrip()
+
+        if not external_sub_guid:
+            logger.error("Could not find public network, please run the " +
+                         "sanity test to create the appropriate networks " +
+                         "and re-run this script with " +
+                         "--tempest_config_only flag.")
+            pass
+
+        self._backup_tempest_conf()
+
         cmds = [
-            'source ~/' + self.settings.overcloud_name + 'rc;'
-            "sudo ip route add " + self.settings.floating_ip_network +
+            'source ~/' + setts.overcloud_name + 'rc;'
+            "sudo ip route add " + setts.floating_ip_network +
             " dev eth0",
-            'source ~/' + self.settings.overcloud_name + 'rc;' +
-            'tempest init mytempest;cd mytempest;' +
+            'source ~/' + setts.overcloud_name + 'rc;' +
+            'tempest init ' + TEMPEST_DIR + ';cd ' + TEMPEST_DIR + ';' +
             'discover-tempest-config --deployer-input ' +
             '~/tempest-deployer-input.conf --debug --create --network-id ' +
             "`openstack subnet list  | grep external_sub | awk " +
             "'{print $6;}'` " +
             "object-storage-feature-enabled.discoverability False",
             'sed -i "s|tempest_roles =.*|tempest_roles = _member_,Member|" ' +
-            '~/mytempest/etc/tempest.conf',
+            TEMPEST_CONF_PATH,
         ]
         for cmd in cmds:
             self.run_tty(cmd)
 
     def run_tempest(self):
-        logger.debug("running tempest")
+        logger.debug("Running tempest")
+
+        if not self._is_tempest_conf():
+            self.configure_tempest()
+
+        os._exit(0)
```

Comment:
There's definitely something wrong here.  We exit unconditionally, and there is a bunch of code following that won't get executed.